.github/workflows/ci_tests.yaml

```diff
@@ -138,3 +141,66 @@ jobs:
           file: ./coverage.xml # optional
           env_vars: OS,PYTHON
           fail_ci_if_error: true
+
+      - name: Checkout the gh-pages-testing branch
+        uses: actions/checkout@28c7f3d2b5162b5ddd3dfd9a45aa55eaf396478b
+        with:
+          ref: gh-pages-testing
+          # Checkout to this folder instead of the current one
+          path: deploy
+          # Download the entire history
+          fetch-depth: 0
+        if: (matrix.os == 'ubuntu-latest') && (matrix.python-version == '3.9')
+        #if: (github.event_name == 'release' || github.event_name == 'push') && (matrix.os == 'ubuntu-latest') && (matrix.python-version == '3.9')
```

Comment:
The `if` condition is a little complicated here.

We don't want to deploy documentation for PRs, thus we limit the triggering event to `release` or `push`.

We don't want to deploy the same documentation multiple times, so we check if we're using Ubuntu + Python 3.9. 

Need to use the commented `if` condition before merging.